DEVELOPMENT.md

```diff
@@ -141,83 +156,102 @@ If you are interested in proposing a new feature, make sure to create a new feat
    ```bash
    git push
    ```
-</details>
+   </details>
 
 <details><summary><h3>using VS Code</h3></summary>
 
 1. Switch to the main branch:
-    1. Open the command palette with Ctrl+Shift+P.
-    2. Search for 'Git: Checkout to...'
-    3. Select 'main'.
+
+   1. Open the command palette with Ctrl+Shift+P.
+   2. Search for 'Git: Checkout to...'
+   3. Select 'main'.
 
 2. Pull from the upstream remote:
-    1. Open the command palette.
-    2. Enter and select 'Git: Pull...'
-    3. Select 'upstream'.
+
+   1. Open the command palette.
+   2. Enter and select 'Git: Pull...'
+   3. Select 'upstream'.
 
 3. Create and change to a new branch:
-    1. Type in 'Git: Create Branch...' in the command palette.
-    2. Enter a branch name.
+
+   1. Type in 'Git: Create Branch...' in the command palette.
+   2. Enter a branch name.
 
 4. Make your changes!
 
 5. Stage all your changes:
-    1. Enter and select 'Git: Stage All Changes...' in the command palette.
+
+   1. Enter and select 'Git: Stage All Changes...' in the command palette.
 
 6. Commit your changes:
-    1. Open the command palette and enter 'Git: Commit'.
+
+   1. Open the command palette and enter 'Git: Commit'.
 
 7. Push your changes:
-    1. Enter and select 'Git: Push...' in the command palette.
+   1. Enter and select 'Git: Push...' in the command palette.
 
 </details>
 
-
 ## Run BentoML with verbose/debug logging
 
 To view internal debug loggings for development, set the `BENTOML_DEBUG` environment variable to `TRUE`:
+
 ```bash
 export BENTOML_DEBUG=TRUE
 ```
 
 And/or use the `--verbose` option when running `bentoml` CLI command, e.g.:
+
 ```bash
 bentoml get IrisClassifier --verbose
 ```
 
 ## Style check, auto-formatting, type-checking
 
-formatter: [black](https://github.com/psf/black), [isort](https://github.com/PyCQA/isort)
+formatter: [black](https://github.com/psf/black), [isort](https://github.com/PyCQA/isort), [buf](https://github.com/bufbuild/buf)
 
-linter: [pylint](https://pylint.org/)
+linter: [pylint](https://pylint.org/), [buf](https://github.com/bufbuild/buf)
 
 type checker: [pyright](https://github.com/microsoft/pyright)
 
+We are using [buf](https://github.com/bufbuild/buf) for formatting and linting
+of our proto files. Configuration can be found [here](./bentoml/grpc/buf.yaml).
+Currently, we are running `buf` with docker, hence we kindly ask our developers
+to have docker available. Docker installation can be found [here](https://docs.docker.com/get-docker/).
+
 Run linter/format script:
+
 ```bash
 make format
 
 make lint
 ```
 
 Run type checker:
+
 ```bash
 make type
 ```
 
+## Editing proto files
+
+The proto files for the BentoML gRPC service are located under [`bentoml/grpc`](./bentoml/grpc/).
+The generated python files are not checked in the git repository, and are instead generated via this [`script`](./scripts/generate_grpc_stubs.sh).
```

Comment:
Why are we using a shell script instead of `setup.py` now?

Reply:
because I ran into way too much trouble with grpcio and setuptools and readthedocs error. I discussed this with Chaoyu and decided that probably just have a script and ask developer to generate it before should be good enough. 

For distribution, we would have to generate the stubs before and include those into the distribution.